src/com/Jacksonnn/DCCore/BannedWords/BannedWordsListener.java

```diff
@@ -2,25 +2,30 @@
 
 import com.Jacksonnn.DCCore.Configuration.ConfigManager;
 import com.Jacksonnn.DCCore.GeneralMethods;
+import org.bukkit.ChatColor;
 import org.bukkit.entity.Player;
 import org.bukkit.event.EventHandler;
 import org.bukkit.event.Listener;
 import org.bukkit.event.player.AsyncPlayerChatEvent;
 
+import java.util.ArrayList;
+import java.util.List;
+
 public class BannedWordsListener implements Listener {
     @EventHandler
     public void onRegularChat(AsyncPlayerChatEvent event) {
         Player player = event.getPlayer();
 
-        for (String word : event.getMessage().split(" ")) {
-            for (String bannedWord : ConfigManager.defaultConfig.get().getStringList("AntiCurse.bannedWords")) {
-                if (word.equalsIgnoreCase(bannedWord)) {
-                    if (player.hasPermission("DCCore.AntiCurse.bypass")) {
-                        return;
-                    } else {
-                        event.setCancelled(true);
-                        event.getPlayer().sendMessage(GeneralMethods.prefix + " Please rethink your choice of words... (don't cuss!)");
-                    }
+        bannedWords = ConfigManager.bannedWords.get().getStringList("AntiCurse.bannedWords");
+
+        for (String bannedWord : bannedWords) {
+            if (event.getMessage().toUpperCase().contains(bannedWord.toUpperCase()) || player.getName().toUpperCase().contains(bannedWord.toUpperCase()) || player.getDisplayName().toUpperCase().contains(bannedWord.toUpperCase())) {
+                if (player.hasPermission("DCCore.AntiCurse.bypass")) {
+                    return;
+                } else {
+                    event.setCancelled(true);
+                    event.getPlayer().sendMessage(GeneralMethods.prefix + " Please rethink your choice of words... (check your username, nickname, or chat message!!!)");
+                    event.getPlayer().sendMessage(ChatColor.YELLOW + "Banned word used: " + bannedWord);
```

Comment:
If a player sends a message with multiple banned words, they receive the "Please rethink your choice of words..." message multiple times. Consider naming a String[] variable (initialize it to null) before entering the loop of bannedWords. If a bannedWord is detected, if the String[] is not initialized, initialize it, then add the word to the String[]. At the end of the loop, if String[] is not null, send the message and cancel the event. Also, consider testing for the DCCore.AntiCurse.bypass permission outside the loop